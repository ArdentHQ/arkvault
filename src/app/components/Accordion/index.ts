export * from "./Accordion.blocks";
export * from "./Accordion.contracts";
