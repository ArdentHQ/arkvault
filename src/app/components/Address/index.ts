export * from "./Address";
export * from "./AddressLink";
