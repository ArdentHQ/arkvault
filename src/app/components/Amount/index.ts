export { Amount } from "./Amount";
export { AmountLabel } from "./AmountLabel";
