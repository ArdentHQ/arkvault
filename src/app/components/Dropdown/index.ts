export * from "./Dropdown";
export * from "./Dropdown.contracts";
