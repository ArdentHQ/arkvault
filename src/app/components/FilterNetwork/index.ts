export * from "./FilterNetwork";
export * from "./FilterNetwork.contracts";
export * from "./NetworkOptions";
export * from "./ToggleAllOption";
