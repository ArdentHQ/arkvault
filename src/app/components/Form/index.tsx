export * from "./Form";
export * from "./Form.contracts";
export * from "./FormButtons";
export * from "./FormField";
export * from "./FormHelperText";
export * from "./FormLabel";
export * from "./SubForm";
