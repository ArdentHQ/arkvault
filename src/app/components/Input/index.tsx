export * from "./Input";
export * from "./InputAddress";
export * from "./InputCounter";
export * from "./InputCurrency";
export * from "./InputDate";
export * from "./InputDefault";
export * from "./InputPassword";
