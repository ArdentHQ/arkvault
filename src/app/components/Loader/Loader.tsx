import React from "react";
import cn from 'classnames';
import { Spinner } from '../Spinner';

export const Loader = ({
    text,
    className,
}: {
    text: string;
    className?: string;
}) => {
    return (
      <div className={cn("w-full px-6 py-5 border border-theme-warning-200 bg-theme-warning-50 dark:bg-transparent dark:border-theme-warning-600 flex flex-row gap-3 rounded-xl items-center", className)} data-testid="Loader__wrapper">
        <Spinner color='warning-alt' size='sm' className='rounded-full' width={3}/>
        <hr className='h-5 w-px border-transparent bg-theme-warning-200 dark:bg-theme-secondary-800' />
        <span className='font-semibold md:text-base text-sm text-theme-secondary-700 dark:text-theme-warning-600' data-testid="Loader__text">{text}</span>
      </div>
    )
  }