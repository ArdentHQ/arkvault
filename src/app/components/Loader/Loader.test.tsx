import React from "react";

import { render, screen } from "@/utils/testing-library";
import { Loader } from '@/app/components/Loader';

describe("Loader", () => {
	it("should render with custom text", () => {
		const { asFragment } = render(<Loader text="Testing" />);

		expect(screen.getByTestId("Loader__wrapper")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

    it("should render with custom class names", () => {
        const { asFragment } = render(<Loader text="Testing" className="custom-class" />);

        expect(screen.getByTestId("Loader__wrapper")).toHaveClass("custom-class");
        expect(asFragment()).toMatchSnapshot();
    })

    it("should render custom text", () => {
        render(<Loader text="Testing" />);
        expect(screen.getByText("Testing")).toBeInTheDocument();
    })
});
