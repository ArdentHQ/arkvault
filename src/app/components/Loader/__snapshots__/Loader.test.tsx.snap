// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader > should render with custom class names 1`] = `
<DocumentFragment>
  <div
    class="flex w-full flex-row items-center gap-3 rounded-xl border border-theme-warning-200 bg-theme-warning-50 px-6 py-5 dark:border-theme-warning-600 dark:bg-transparent custom-class"
    data-testid="Loader__wrapper"
  >
    <div
      class="animate-spin border-[5px] flex-shrink-0 w-5 h-5 border-theme-warning-200 dark:border-theme-secondary-800 border-l-theme-warning-900 dark:border-l-theme-warning-600 rounded-full"
    />
    <hr
      class="h-5 w-px border-transparent bg-theme-warning-200 dark:bg-theme-secondary-800"
    />
    <span
      class="text-sm font-semibold text-theme-secondary-700 dark:text-theme-warning-600 md:text-base"
      data-testid="Loader__text"
    >
      Testing
    </span>
  </div>
</DocumentFragment>
`;

exports[`Loader > should render with custom text 1`] = `
<DocumentFragment>
  <div
    class="flex w-full flex-row items-center gap-3 rounded-xl border border-theme-warning-200 bg-theme-warning-50 px-6 py-5 dark:border-theme-warning-600 dark:bg-transparent"
    data-testid="Loader__wrapper"
  >
    <div
      class="animate-spin border-[5px] flex-shrink-0 w-5 h-5 border-theme-warning-200 dark:border-theme-secondary-800 border-l-theme-warning-900 dark:border-l-theme-warning-600 rounded-full"
    />
    <hr
      class="h-5 w-px border-transparent bg-theme-warning-200 dark:bg-theme-secondary-800"
    />
    <span
      class="text-sm font-semibold text-theme-secondary-700 dark:text-theme-warning-600 md:text-base"
      data-testid="Loader__text"
    >
      Testing
    </span>
  </div>
</DocumentFragment>
`;
